rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -420,8 +425,33 @@ public boolean createNodeSource(@HeaderParam("sessionid") String sessionId,
                 }
             }
         }
-        return rm.createNodeSource(nodeSourceName, infrastructureType, infraParams, policyType, policyParams)
-                 .getBooleanValue();
+        try {
+            nsState.setResult(rm.createNodeSource(nodeSourceName,
+                                                  infrastructureType,
+                                                  infraParams,
+                                                  policyType,
+                                                  policyParams)
+                                .getBooleanValue());
+            return nsState;
+        } catch (RuntimeException ex) {
+            nsState.setResult(false);
+            nsState.setErrorMessage(cleanDisplayedErrorMessage(ex.getMessage()));
+            nsState.setStackTrace(getStackTrace(ex));
+            return nsState;
+        }
+
+    }
+
+    private String cleanDisplayedErrorMessage(String errorMessage) {
+        String regex = "^[^:]*:(.*)";
+        Pattern pattern = Pattern.compile(regex);
```

Comment:
This pattern can be created only one time at constructor level

Reply:
pattern should be private static final